Add Computational Intelligence in Sports

alenrajsp opened the pull request:
**https://github.com/firefly-cpp/awesome-computational-intelligence-in-sports#readme**

This is a list of resources related to Computational Intelligence (CI) methods in sports. The list includes books, scientific literature, datasets, software, and web applications that are relevant to researchers who work on applying CI techniques in sports.

The list is curated to provide researchers with easy access to a wide range of resources on the topic. The resources in the list are cited using the MLA 8th edition format and cover a variety of sports such as cycling, soccer, track and field, triathlon, and wrestling.

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- [✔️] Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it. **Instead use [#2242](https://github.com/sindresorhus/awesome/issues/2242) for incubation visibility**.
- [✔️] **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- [✔️] **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).**
	Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration. **Just commenting “looks good” or simply marking the pull request as approved does not count!** You have to actually point out mistakes or improvement suggestions. Comments pointing out lint violation are allowed, but does **not** count as a review.
    - #2560 -> [Comment](https://github.com/sindresorhus/awesome/issues/2560#issuecomment-1485589406_)
    - #2550 -> [Comment](https://github.com/sindresorhus/awesome/issues/2550#issuecomment-1480877823_)
- [✔️] You have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/main/create-list.md).
- [✔️] This pull request has a title in the format `Add Name of List`. It should not contain the word `Awesome`.
	- ✅ `Add Swift`
	- ✅ `Add Software Architecture`
	- ❌ `Update readme.md`
	- ❌ `Add Awesome Swift`
	- ❌ `Add swift`
	- ❌ `add Swift`
	- ❌ `Adding Swift`
	- ❌ `Added Swift`
- [✔️] Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb.
	- ✅ `- [iOS](…) - Mobile operating system for Apple phones and tablets.`
	- ✅ `- [Framer](…) - Prototyping interactive UI designs.`
	- ❌ `- [iOS](…) - Resources and tools for iOS development.`
	- ❌ `- [Framer](…)`
	- ❌ `- [Framer](…) - prototyping interactive UI designs`
- [✔️] Your entry should be added at the bottom of the appropriate category.
- [✔️] The title of your entry should be title-cased and the URL to your list should end in `#readme`.
	- Example: `- [Software Architecture](https://github.com/simskij/awesome-software-architecture#readme) - The discipline of designing and building software.`
- [✔️] The suggested Awesome list complies with the below requirements.

## Requirements for your Awesome list

- [✔️] **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- [✔️] Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please [report it](https://github.com/sindresorhus/awesome-lint/issues/new).
- [✔️] The default branch should be named [`main`, not `master`](https://www.zdnet.com/article/github-to-replace-master-with-alternative-term-to-avoid-slavery-references/).
- [✔️] **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- ✅ `Mobile operating system for Apple phones and tablets.`
	- ✅ `Prototyping interactive UI designs.`
	- ❌ `Resources and tools for iOS development.`
	- ❌ `Awesome Framer packages and tools.`
- [✔️] It's the result of hard work and the best I could possibly produce.
	**If you have not put in considerable effort into your list, your pull request will be immediately closed.**
- [✔️] The repo name of your list should be in lowercase slug format: `awesome-name-of-list`.
	- ✅ `awesome-swift`
	- ✅ `awesome-web-typography`
	- ❌ `awesome-Swift`
	- ❌ `AwesomeWebTypography`
- [✔️] The heading title of your list should be in [title case](https://capitalizemytitle.com/) format: `# Awesome Name of List`.
	- ✅ `# Awesome Swift`
	- ✅ `# Awesome Web Typography`
	- ❌ `# awesome-swift`
	- ❌ `# AwesomeSwift`
- [✔️] Non-generated Markdown file in a GitHub repo.
- [✔️] The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- [✔️] Not a duplicate. Please search for existing submissions.
- [✔️] Only has awesome items. Awesome lists are curations of the best, not everything.
- [✔️] Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs. If you really need to include such items, they should be in a separate Markdown file.
- [✔️] Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- **The image should be high-DPI.** Set it to maximum half the width of the original image.
- [✔️] Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- [✔️] Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/main/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- [✔️] Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
	- Must not feature `Contributing` or `Footnotes` sections.
- [✔️] Has an appropriate license.
	- **We strongly recommend the [CC0 license](https://creativecommons.org/publicdomain/zero/1.0/), but any [Creative Commons license](https://creativecommons.org/choose/) will work.**
		- Tip: You can quickly add it to your repo by going to this URL: `https://github.com/<user>/<repo>/community/license/new?branch=main&template=cc0-1.0` (replace `<user>` and `<repo>` accordingly).
	- A code license like MIT, BSD, Apache, GPL, etc, is not acceptable. Neither are WTFPL and [Unlicense](https://unlicense.org).
	- Place a file named `license` or `LICENSE` in the repo root with the license text.
	- **Do not** add the license name, text, or a `Licence` section to the readme. GitHub already shows the license name and link to the full text at the top of the repo.
	- To verify that you've read all the guidelines, please comment on your pull request with just the word `unicorn`.
- [✔️] Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/main/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
	- It can optionally be linked from the readme in a dedicated section titled `Contributing`, positioned at the top or bottom of the main content.
	- The section should not appear in the Table of Contents.
- [✔️] All non-important but necessary content (like extra copyright notices, hyperlinks to sources, pointers to expansive content, etc) should be grouped in a `Footnotes` section at the bottom of the readme. The section should not be present in the Table of Contents.
- [✔️] Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- [✔️] Does not use [hard-wrapping](https://stackoverflow.com/questions/319925/difference-between-hard-wrap-and-soft-wrap).
- [✔️] Does not include a CI (e.g. GitHub Actions) badge.<br>You can still use a CI for linting, but the badge has no value in the readme.
- [✔️] Does not include an `Inspired by awesome-foo` or `Inspired by the Awesome project` kinda link at the top of the readme. The Awesome badge is enough.

**Go to the top and read it again.**

alenrajsp commented:
unicorn

thoughtspile commented:
One minor thing, not sure if the license badge fits this rule:

> Do not add the license name, text, or a Licence section to the readme. GitHub already shows the license name and link to the full text at the top of the repo.

rickstaa commented:
The list looks good to me. The only things I was not sure about are the license (similar to @thoughtspile's comment) and the fact that you don't use bullet points for your articles. 

![image](https://user-images.githubusercontent.com/17570430/229353605-b565328e-c883-4bbd-bdc4-243e512fae1a.png)

There doesn't seem guildines yet for adding articles so we should check with @sindresorhus.

alenrajsp commented:
@thoughtspile Thank you. We've removed the badge in the newest version.

@rickstaa Thanks for the feedback. I've added bullet points to the articles. I however am not sure regarding changing the format since the mentioned https://github.com/vhladiienko/awesome-game-production#-articles repo contains informal articles and here in our repository there is only scientific peer-reviewed literature.

alenrajsp commented:
🦄  :)

alenrajsp commented:
Please let me know if anyone has any other additions or comments on how to improve this repository.

alenrajsp commented:
unicorn

alenrajsp commented:
Also could anyone check why the ToC gives error. Thank you.

thoughtspile commented:
@alenrajsp looks like you have an extra 📚 emoji in the section title, but not in the ToC

alenrajsp commented:
I have fixed the ordering so that it is consistent (e.g. <TITLE> <EMOJI>) for both the ToC and actual section titles.
I however still have a big problem with the linter. Let me explain
I have a fork of our repository so disregard the fork doesn't have awesome tags.

This is the fork link: https://github.com/alenrajsp/comp-sport-literature/tree/main

Running the command: npx awesome-lint https://github.com/alenrajsp/comp-sport-literature/
Outputs the following message:
![image](https://github.com/sindresorhus/awesome/assets/27721714/dc214977-dd12-4c0a-ad68-051546f33ebc)

I have no idea how to fix error at lines 24:3 because if I remove the %EF%... (unicode sign for a space) the ToC wont work. You can test the link yourself and see that it works, the linter however doesn't recognise it as a valid link. Does anyone have any idea what could I do?

sindresorhus commented on readme.md:
```diff
@@ -504,6 +504,7 @@
 - [Theoretical Computer Science](https://github.com/mostafatouny/awesome-theoretical-computer-science#readme) - The interplay of computer science and pure mathematics, distinguished by its emphasis on mathematical rigour and technique.
 - [Conversational AI](https://github.com/jyguyomarch/awesome-conversational-ai#readme) - Build awesome chatbots and digital assistants.
 - [Generative AI](https://github.com/steven2358/awesome-generative-ai#readme) - Automatically generate a wide range of unique content in text, image and audio format.
+- [Computational Intelligence in Sports](https://github.com/firefly-cpp/awesome-computational-intelligence-in-sports#readme) - Research, methods, datasets, and software related to use of Computational Intelligence in Sports.
```
> Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.**

alenrajsp replied:
I am sorry for the a bit longer reply since I was on holidays. I have fixed the description according to your instructions in [this commit](https://github.com/sindresorhus/awesome/pull/2565/commits/4bb471e2cf0e48f602cf53db1d3ba89b52902eac). I believe the repo is now fixed, but I believe the check will still fail because the Linder does _not like_ emojis, although the ToC works correctly.

sindresorhus commented:
I think this list would be a lot more useful if the research paper entries had a short description. You could have ChatGPT generate this based on the paper abstract/summary.

sindresorhus commented:
While the academic reference formatting works in academic settings, it makes it really hard to go through the list. I strongly recommend using the format:

```
- [Title]() - Description.
```

You could put the academic reference part on a separate line

```
- [Population-Based Metaheuristics for Planning Interval Training Sessions in Mountain Biking]() - Description.
  Fister Jr, Iztok, et al. Lecture Notes in Computer Science (Including Subseries Lecture Notes in Artificial Intelligence and Lecture Notes in Bioinformatics), vol. 11655 LNCS, Springer, Cham, 2019, pp. 70–79, doi:10.1007/978-3-030-26369-0_7.
```
